test/smtcallback.js

```diff
@@ -132,8 +132,10 @@ tape('SMTCheckerCallback', function (t) {
       return;
     }
 
-    if (smtsolver.availableSolvers === 0) {
-      st.skip('No SMT solver available.');
+    // For these tests we actually need z3/Spacer.
+    const z3HornSolvers = smtsolver.availableSolvers.filter(solver => solver.command === 'z3');
+    if (z3HornSolvers.length === 0) {
```

Comment:
Not to change things too much, we require z3 to run these tests.

Reply:
I think we should make it opt-out instead of opt-in. With some equivalent of the `--no-smt` flag. Otherwise it's too easy for such tests to never run in CI.